internal/modules/modules.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: This file implements fetching modules for a given
+// service manifest.
+
+// Package modules implements all logic needed for interacting
+// with stencil modules and their interaction with a service generated
+// by stencil.
+package modules
+
+import (
+	"context"
+	"path"
+
+	"github.com/getoutreach/stencil/pkg/configuration"
+	"github.com/pkg/errors"
+	giturls "github.com/whilp/git-urls"
+)
+
+// GetModulesForService returns a list of modules that a given service manifest
+// depends on. They are not returned in the order of their import.
+func GetModulesForService(ctx context.Context, m *configuration.ServiceManifest) ([]*Module, error) {
+	// We only support importing a single version of a module
+	// IDEA(jaredallard): Always use the latest?
```

Comment:
we should definitely pin versions - we talked about this a bit in standup the other day

Reply:
Yeah, this comment was to myself to remove before merging. That's just what to do when a dep is required by the same package @ different versions